addons/hr_holidays/tests/test_accrual_requests.py

```diff
@@ -0,0 +1,184 @@
+            })
+
+    @mute_logger('odoo.models.unlink', 'odoo.addons.mail.models.mail_mail')
+    def test_accrual_enough_days_left(self):
+        """  Employee creates a leave request in a limited category and has enough days left  """
+        aloc1_user_group = self.env['hr.leave.allocation'].sudo(self.user_hruser_id).create({
+            'name': 'Days for limited category',
+            'employee_id': self.employee_emp_id,
+            'holiday_status_id': self.holidays_type_2.id,
+            'number_of_days': 2,
+        })
+        aloc1_user_group.action_approve()
+
+        holiday_status = self.holidays_type_2.sudo(self.user_employee_id)
+        self._check_holidays_status(holiday_status, 2.0, 0.0, 2.0, 2.0)
+
+        hol = self.env['hr.leave'].sudo(self.user_employee_id).create({
+            'name': 'Hol11',
+            'employee_id': self.employee_emp_id,
+            'holiday_status_id': self.holidays_type_2.id,
+            'date_from': (datetime.today() - relativedelta(days=2)),
+            'date_to': datetime.today(),
+            'number_of_days': 2,
+        })
+
+        holiday_status.invalidate_cache()
+        self._check_holidays_status(holiday_status, 2.0, 0.0, 2.0, 0.0)
+
+        hol.sudo(self.user_hrmanager_id).action_approve()
+
+        self._check_holidays_status(holiday_status, 2.0, 2.0, 0.0, 0.0)
+
+    @mute_logger('odoo.models.unlink', 'odoo.addons.mail.models.mail_mail')
+    def test_accrual_hr_manager_not_enough_days_left(self):
+        """  Employee creates a leave request in a limited category and has enough days left.
+        After a HR manager try to modify the number of days but has not enough days """
+        aloc1_user_group = self.env['hr.leave.allocation'].sudo(self.user_hruser_id).create({
+            'name': 'Days for limited category',
+            'employee_id': self.employee_emp_id,
+            'holiday_status_id': self.holidays_type_2.id,
+            'number_of_days': 2,
+        })
+        aloc1_user_group.action_approve()
+
+        hol = self.env['hr.leave'].sudo(self.user_employee_id).create({
+            'name': 'Hol11',
+            'employee_id': self.employee_emp_id,
+            'holiday_status_id': self.holidays_type_2.id,
+            'date_from': (datetime.today() - relativedelta(days=1)),
+            'date_to': datetime.today(),
+            'number_of_days': 1,
+        })
+
+        hol.sudo(self.user_hrmanager_id).action_draft()
+
+        # try to change to 4 days, less than available
+        hol.sudo(self.user_hrmanager_id).write({
+            'date_from': (datetime.today() + relativedelta(days=4)),
+            'date_to': (datetime.today() + relativedelta(days=7)),
+            'number_of_days': 4,
+        })
+        with self.assertRaises(ValidationError):
+            hol.sudo(self.user_hrmanager_id).action_confirm()
+
+    @mute_logger('odoo.models.unlink', 'odoo.addons.mail.models.mail_mail')
+    def test_accrual_validity_time_valid(self):
```

Comment:
I suggest renaming this to `test_type_validity` and merging it with the next one